legal-api/src/legal_api/resources/v2/internal_services.py

```diff
@@ -36,39 +36,40 @@ def update_bn_move():
     if not json_input or not (old_bn := json_input.get("oldBn")) or not (new_bn := json_input.get("newBn")):
         return ({"message": "No oldBn or newBn in body of post."}, HTTPStatus.BAD_REQUEST)
 
-    legal_entity = LegalEntity.find_by_tax_id(old_bn)
-    if legal_entity:
-        legal_entity.tax_id = new_bn
-        legal_entity.save()
+    for identifier in json_input.keys():
+        business = business_service.fetch_business(identifier)
+        if business and business.is_legal_entity:
+            business.tax_id = json_input[identifier]
+            business.save()
```

Comment:
i'm not sure about this code.  

Why do we need to loop through the keys?  Have the implementation change in `main`?

Reply:
Nothing has changed